challenges/easy/caesar-cipher/solutions/valentinadagata/main.js

```diff
@@ -0,0 +1,69 @@
+const letters = 26;
+
+function encode(str, shift) {
+  shift  = shift % letters
+
+  return str.split("").map(character => {
+      // if is a letter
+      if ((character >= "a" && character <= "z") || (character >= "A" && character <= "Z")) {
+        // if is lowercase
+        if (character >= "a" && character <= "z") {
+          ascii = 97
+        }
+        // if is uppercase
+        else {
+          ascii = 65
+        }
+
+        new_str = letters + character.charCodeAt(0) - ascii
+        new_str = (new_str + shift) % letters
+        encrypt = String.fromCharCode(new_str + ascii) 
+        return encrypt
+      }
+      // don't encrypt if it's not a letter
+      else return character;
+  }).join("");
+}
+
+
+function decode(str, shift) {
```

Comment:
Here you're repeating most of the `encode` function. Is there a way to avoid all those repetitions?